lib/seam/request.rb

```diff
@@ -54,9 +44,9 @@ def handle_error_response(response, _method, _uri)
         data: error["data"]
       }
 
-      raise SeamHttpInvalidInputError.new(error_details, status_code, request_id) if error_type == "invalid_input"
+      raise Errors::SeamHttpInvalidInputError.new(error_details, status_code, request_id) if error_type == "invalid_input"
 
-      raise SeamHttpApiError.new(error_details, status_code, request_id)
+      raise Errors::SeamHttpApiError.new(error_details, status_code, request_id)
```

Comment:
Currently they would do `Seam:Errors:SeamHttpApiError`. We can `include Errors` in the `Seam` module and then they would be able to do both `Seam:Errors:SeamHttpApiError` and `Seam::SeamHttpApiError`.

`Seam::HttpApiError` would indeed be more idiomatic than `Seam::SeamHttpApiError`. The Seam prefix seems a bit redundant since it's already in the namespace.
For example, [Stripe ruby gem](https://github.com/stripe/stripe-ruby/blob/master/lib/stripe/errors.rb) doesn't seem to prefix errors with `Stripe`.

Reply:
The only prefixed Stripe error is `StripeError` which is the base error which other specific errors inherit from [[stripe docs ref](https://docs.stripe.com/api/errors/handling?lang=ruby)]

Reply:
OAuth specific errors stripe puts under [OAuth namespace](https://github.com/stripe/stripe-ruby/blob/431140523dc8df165aa2730daf3a133d8ce16808/lib/stripe/errors.rb#L120) e.g. `Stripe::OAuth::InvalidClientError`

Reply:
For ex. could do `Seam::Http::ApiError` and `Seam::ActionAttempt::TimeoutError`?

Reply:
> For ex. we could do `Seam::Http::ApiError` and `Seam::ActionAttempt::TimeoutError`? Though, I'm not sure about this.

Ok yes, this makes sense! Having `Errors` is grouping things by type, but really we should groups things by scope, i.e., a module should have everything it needs to work and not need to reach outside.

Since with Ruby, Seam is a module and not a class, how to construct an instance?

```
Seam.new(api_key) # or Seam::Http.new ?
Seam.from_api_key(api_key)
Seam::Webhooks.new
```

Reply:
Currently to construct an instance you would

```rb
Seam::Client.new(api_key) # that's what current SDK supports https://github.com/seamapi/ruby?tab=readme-ov-file#example
Seam::Client.from_api_key("some-invalid-key-format")
Seam::ClientMultiWorkspace.from_personal_access_token(personal_access_token)
```

Reply:
Is there anyway we can get `Seam.new` to work?

I know this comment thread is not in scope of the PR but we do need to sort out the top level API :) 

`Seam::Client` should be reserved for the underlying client, we can call this `Seam::Http` instead and `Seam::MultiWorkspaceHttp`.

Reply:
Yea, we could add a factory method to Seam module in the main file `lib/seam.rb` like so:
```rb
module Seam
  def self.new(**args)
    Client.new(**args)
  end
end
```
This way we can use `Seam.new(api_key: "...")` and also as a namespace `Seam::Errors::HttpApiError`.

Reply:
Ruby allows for very expressive libraries. Let's hide the internals from the users and build on the `Seam` module.

So,


```ruby
module Seam
  def self.new(**args)
    Http.new(**args)
  end

  def from_api_key
  end
  
  def from_personal_access_token
  end

  # Expose lts_version and version here
end
```

Errors are `Seam::HttpApiError` and `Seam::ActionAttemptTimeoutError`, etc as those are shared by things in the module.

The invalid option errors should be class level though, so `Seam:Http:InvalidOptionsError` (these errors are internal, the user is not meant to really catch them and handle them).

The other classes must be accessed directly, e.g., `Seam::Webhook.new` and `Seam::HttpMultiWorkspace.new`.

Reply:
Sounds good! I'll do the changes in a separate RP

Reply:
https://github.com/seamapi/ruby-next/issues/71